src/org/thoughtcrime/securesms/NicknameHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.RecipientDatabase;
+import org.thoughtcrime.securesms.recipients.Recipient;
+
+
+public class NicknameHandler {
+    private Context context;
+    private RecipientDatabase recipientDatabase;
+
+    public NicknameHandler(Context context) {
+        this.context = context;
+    }
+
+    public NicknameHandler setContext(Context context) {
+        this.context = context;
+        return this;
+    }
+
+    /**
+     * This method allows the developer to set the underlining database
+     * to any kind of database object passed in the parameter
+     * This method is implemented for only testing purposes inorder
+     * to pass mocked database object
+     * @param recipientDatabase
+     * @return
+     */
+    public NicknameHandler setRecipientDatabase(RecipientDatabase recipientDatabase) {
+        this.recipientDatabase = recipientDatabase;
+        return this;
+    }
+
+    /**
+     * This method responsible for setting the nick name
+     * of a recipient. If the new nickname is identical to the
+     * old one, then it will return false, otherwise, the nickname
+     * will be updated and this will return true.
+     * @param recipient
+     * @param nickname
+     * @return boolean indicating whether the nickname was set or not
+     */
+    public boolean setNickname(Recipient recipient, String nickname) {
+        setupDatabaseHandler();
+        return recipientDatabase.setNickname(recipient, nickname);
+    }
+
+    // TODO by DAN
+    public boolean removeNickname(Recipient recipient) {
+        setupDatabaseHandler();
+
+        // TODO call the correct remove method
```

Comment:
This means we cannot merge yet?

Reply:
We can merge, actuall @Dan-Yoo  needs to continue working from this code from his branch

Reply:
So remove this from your branch then ?

Reply:
I think it is nice to keep it to guide him and he ll know what is missing and from where he wants to start. But maybe we can rename this branch to "nickname_backend" and will not be merged until everything is finished.. @Dan-Yoo what do you think ?